examples/metrics/application_efficiency.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: 0BSD
+"""
+Working with Application Efficiency
+===================================
+
+Understanding relative performance.
+
+One goal of P3 analysis is to understand how well a given application is able
+to adapt to and make effective use of the capabilities of different
+platforms. Comparisons of *raw* performance (e.g., time to solution) across
+platforms can't help us, because raw performance doesn't reflect how fast an
+application **should** run.
+
+To address this, the P3 Analysis Library works with normalized *performance
+efficiency* data. With normalized data, an application's performance can be
+represented as a number in the range :math:`[0, 1]`, where :math:`1` means
+an application is achieving the best possible performance.
+
+Application Efficiency
+----------------------
+
+The simplest form of performance efficiency we can work with is *application
+efficiency*, defined below:
+
+**Application Efficiency**
+  The performance of an *application*, measured relative to the best known
+  performance previously demonstrated for solving the same *problem* on the
+  same *platform*.
+
+Working with application efficiency is simple because it does not rely on
+performance models or theoretical hardware limits. Although it can't tell us
+whether an application is performing as well as theoretically possible, it
+shows how an application compares to the state-of-the-art, which is often good
+enough.
+
+Calculating Application Efficiency
+----------------------------------
+
+Let's begin with a very simple example, with a single application, using the
+data below:
+
+.. list-table::
+    :widths: 20 20 20 20 20
+    :header-rows: 1
+
+    * - problem
+      - application
+      - platform
+      - fom
+      - date
+
+    * - Test
+      - MyApp
+      - A
+      - 25.0
+      - 2023
+    * - Test
+      - MyApp
+      - B
+      - 12.5
+      - 2023
+    * - Test
+      - MyApp
+      - C
+      - 25.0
+      - 2023
+    * - Test
+      - MyApp
+      - D
+      - NaN
+      - 2023
+    * - Test
+      - MyApp
+      - E
+      - 5.0
+      - 2023
+
+.. tip::
+    A NaN or 0.0 performance result is interpreted by the P3 Analysis Library
+    to mean that an application did not run, or did not not run correctly.
```

Comment:
More generally, could you say that it just didn't pass a unit test being either it didn't run, or it didn't produce a numerical result or something that we expect?

Depending on your target audience, I figured it might be good to qualify what "running correctly" comprises